fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/ReindexManager.java

```diff
@@ -164,37 +164,18 @@ public int getErrorCount() {
 
     /**
      * Index the membership relationships
-     * @throws InterruptedException If fails to succeed after 3 tries.
      */
-    private void indexMembership() throws InterruptedException {
-        // How much to back off between each attempt (seconds).
-        final long backoff = 10L;
-        int attempt = 0;
-        boolean finished = false;
-        while (!finished && attempt < 3) {
-            try {
-                reindexService.indexMembership(transaction());
-                finished = true;
-            } catch (final Exception e) {
-                reindexService.rollbackMembership(transaction().getId());
-            }
-            attempt += 1;
+    private void indexMembership() {
+        try {
+            reindexService.indexMembership(transaction());
+        } catch (final RuntimeException e) {
             try {
-                TimeUnit.SECONDS.sleep(backoff * attempt);
-            } catch (final InterruptedException e) {
-                // Continue on if we are interrupted.
+                reindexService.rollbackMembership(transaction());
```

Comment:
Using a shortlived transaction (line 225 below) means that the only action that can be rolled back by the `ReindexService` is resetting the search index, all others are now skipped. Not saying this is right or wrong, but we could probably remove this whole `try catch`.

Reply:
You're saying remove membership rollback?

Reply:
Doh, I read `rollback` but you removed that method. You can ignore this.